### ISSUE TITLE
Got A Bug When Closing

### ISSUE THREAD

**Issue by ClwnMan76** [arsenetar/dupeguru#857]
First off, sincerest apologies as I've never done this before but such a huge fan of the work you do I figured it was the least I could do!

**Describe the bug**
Gave the following box when trying to close the program. Odd thing is I had to click close several times, report on here, and then  finally the escape key several times before it would finally stop popping up. I don't know if that's normal.

Application Name: dupeGuru
Version: 4.1.0

Traceback (most recent call last):
  File "qt\details_table.py", line 64, in headerData
  File "core\gui\details_panel.py", line 42, in row_count
AttributeError: 'DetailsPanel' object has no attribute '_table'

**To Reproduce**
I was running some tests comparing to another application similar (hasn't been updated in years). So I was running several tests on the same folder. I can't be 100% sure these were the exact steps. I was really busy at the time. My apologies!

1. Ran picture search on folder with about 50 "hardness".
2. After seeing results I went back and lowered the "hardness" to get different results.
3. After seeing these results I simply hit the x to close the program.

**Expected behavior**
Hit the X and the program closes.

**Screenshots**
I'll know for next time!

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Edition	Windows 10 Home
Version	20H2
Installed on	‎10/‎6/‎2020
OS build	19042.867
Experience	Windows Feature Experience Pack 120.2212.551.0

**Additional context**
Only thing in the debug log...

2021-03-17 10:30:40,492 - WARNING - Error while loading C:/Users/Storm/AppData/Local/Hardcoded Software/dupeGuru\exclude_list.xml: [Errno 2] No such file or directory: 'C:/Users/Storm/AppData/Local/Hardcoded Software/dupeGuru\\exclude_list.xml'

I have no programming knowledge, but I'm really good with computers otherwise. If there's any way I can help please let me know!

**Comment by glubsy** [comment]
Thanks for the report! 
Had a quick look but could not reproduce. From what I understand this only happened once, it did not happen several times across different sessions.

That error from the debug log is not related.

I suspect this happens when changing Application Mode between scans but I could be wrong. Need more info.

**Comment by BacchusPS** [comment]
I've been getting this almost every time I shut down the app from picture mode after having done at least one scan, even before 4.1.1. The only way to get out of the crash cycle is to hit esc at least 8-10 times, or just got to Task Manager and shut the process down.

However I think I've finally found the specific and repeatable situation that it happens in. 

Note: I generally only use picture mode, so I always have to switch application modes. Don't know if that's relevant or not. As an aside: It would be nice to be able to set a default mode in the options screen.

My stack trace report was:
Application Name: dupeGuru
Version: 4.1.1

Traceback (most recent call last):
  File "qt\details_table.py", line 64, in headerData
  File "core\gui\details_panel.py", line 42, in row_count
AttributeError: 'DetailsPanel' object has no attribute '_table'

I'm on Windows 7 64 bit. (Yeah, I know .. )

I noticed the DetailsPanel line and figured it was related to that, and it is.

Specific situation/actions:

Possibly relevant:
1) Start DupeGuru
2) Clear Picture Cache
3) Set app mode to Pictures

Definitely Relevant:
4) Start a scan that finds at least one set of duplicates. (It doesn't matter if you add a new directory or scan an existing one.)
5) Turn on Details Panel
6) Click on at least one duplicate image name to bring both the reference and the dupe in the Details Panel.
7) Mark files to remove, including current/last image details panel.
8) Send Marked to Recycle Bin. Again, the dupe in the details panel MUST be one of the files sent to recycle bin.
9) Close the application using the Close (X) Button in the upper right of the app, and NOT file ... quit.

This sequence always results in a crash, and it doesn't matter how many files were recycled. Note that I did not specify closing the details panel in the sequence. If you close that before closing the app, there is no crash. The fact that the details panel is open and has an image on screen that is no longer at the same location on disc appears to be what spawns the error, but only if you use the close button. I have no idea why it matters if you close the DupeGuru via the Close button in the app panel on the upper right, but it does. If you choose file .. quit in the exact same scenario, the app does not spawn the cycle of errors. Only the close button does.

That does at least provide two workarounds to the problem, since either closing the details panel or using file .. quit avoids the crash. Still, it would be nice to not have to remember that, since the details panel should just be discarded int he close process. Whether from file .. quit or the close button.

**Comment by glubsy** [comment]
Seems to be a Microsoft Windows specific bug then. I will need to look into it further and try reproduce in a virtual machine.

**Comment by glubsy** [comment]
Ok I managed to reproduce. This is not Windows-specific after all. It happens whenever the Results table is updated, either by sending an item to the trash, or removing items from the table while the details dialog is floating. No crash if the details dialog is docked for some reason.

Toggling the details dialog off and on again prevents the crash from happening and I don't quite understand why.

Clearing the picture cache doesn't seem to have any incidence on this.

**Comment by 1024mb** [comment]
> 
> 
> Ok I managed to reproduce. This is not Windows-specific after all. It happens whenever the Results table is updated, either by sending an item to the trash, or removing items from the table while the details dialog is floating. No crash if the details dialog is docked for some reason.
> 
> Toggling the details dialog off and on again prevents the crash from happening and I don't quite understand why.
> 
> Clearing the picture cache doesn't seem to have any incidence on this.

I'm having a similar error and I don't have the details window opened at all. It happens to me randomly when I'm deleting duplicates using (I don't know if this has anything to do with it but just mentioning) "directly delete" and "link deleted files", also it happens in both modes, filenames and content/picture.  
After I get the error the whole list behaves weirdly, if I activate both boxes "delta values" and "dups only" some duplicates aren't shown/listed, if I disable those boxes, specifically the "dups only" the missing items return.

```
Application Name: dupeGuru
Version: 4.1.1

Traceback (most recent call last):
  File "qtlib\table.py", line 104, in data
  File "qt\results_model.py", line 43, in _getData
  File "core\gui\result_table.py", line 44, in is_cell_delta
AttributeError: 'NoneType' object has no attribute 'get_display_info'
```

**Comment by glubsy** [comment]
@1024mb I believe this was fixed in #905. If you still encounter the crash you mentioned, please open a new issue ticket as it may be a different issue.
